preview/Win7Msix/Win7MSIXInstaller/StartupTask.cpp

```diff
@@ -0,0 +1,193 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "StartupTask.hpp"
+#include "GeneralUtil.hpp"
+#include "Constants.hpp"
+#include <TraceLoggingProvider.h>
+#include <taskschd.h>
+
+const PCWSTR StartupTask::HandlerName = L"StartupTask";
+
+const PCWSTR StartupTask::TaskDefinitionXmlPrefix =
+    L"<Task xmlns=\"http://schemas.microsoft.com/windows/2004/02/mit/task\">\
+        <Principals>\
+            <Principal id=\"Author\">\
+                <LogonType>InteractiveToken</LogonType>\
+                <RunLevel>LeastPrivilege</RunLevel>\
+            </Principal>\
+        </Principals>\
+        <Triggers>\
+            <LogonTrigger>\
+                <Enabled>true</Enabled>\
+            </LogonTrigger>\
+        </Triggers>\
+        <Settings>\
+            <MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>\
+            <DisallowStartIfOnBatteries>true</DisallowStartIfOnBatteries>\
+            <StopIfGoingOnBatteries>true</StopIfGoingOnBatteries>\
+            <AllowHardTerminate>true</AllowHardTerminate>\
+            <StartWhenAvailable>false</StartWhenAvailable>\
+            <RunOnlyIfNetworkAvailable>false</RunOnlyIfNetworkAvailable>\
+            <AllowStartOnDemand>true</AllowStartOnDemand>\
+            <Enabled>true</Enabled>\
+            <Hidden>false</Hidden>\
+            <RunOnlyIfIdle>false</RunOnlyIfIdle>\
+            <WakeToRun>false</WakeToRun>\
+            <Priority>7</Priority>\
+            <UseUnifiedSchedulingEngine>true</UseUnifiedSchedulingEngine>\
+        </Settings>\
+        <Actions Context=\"Author\">\
+            <Exec>\
+                <Command>\"";
+const PCWSTR StartupTask::TaskDefinitionXmlPostfix =
+                L"\"</Command>\
+            </Exec>\
+        </Actions>\
+    </Task>";
+
+const PCWSTR windowsTaskFolderName = L"\\Microsoft\\Windows";
+const PCWSTR taskFolderName = L"MsixCore";
+
+HRESULT StartupTask::ExecuteForAddRequest()

```

Comment:
Please add comments to the methods to briefly describe the algorithm used to add the task and similarly for the remove case. Also document any known errors that the caller can expect.

Reply:
method summary comments are in the .hpp 

---
In reply to: [277070726](https://github.com/Microsoft/msix-packaging/pull/87#discussion_r277070726) [](ancestors = 277070726)